java/yb-cql/src/test/java/org/yb/cql/SimpleQueryTest.java

```diff
@@ -0,0 +1,540 @@
+
+    Object[][] expected = new Object[10][];
+    for (int t2 = 0; t2 < 10; t2++)
+      expected[t2] = row("key", 15, t2);
+
+    assertRows(execute("SELECT k, t1, t2 FROM %s WHERE k=? AND t1=?", "key", 15), expected);
+
+    Object[][] expectedReverse = new Object[10][];
+    for (int t2 = 9; t2 >= 0; t2--)
+      expectedReverse[9 - t2] = row("key", 15, t2);
+
+    assertRows(execute("SELECT k, t1, t2 FROM %s WHERE k=? AND t1=? ORDER BY t1 DESC, t2 DESC", "key", 15), expectedReverse);
+  }
+
+  @Test
+  public void testRowDeletion() throws Throwable
+  {
+    int N = 4;
+
+    createTable("CREATE TABLE %s (k text, t int, v1 text, v2 int, PRIMARY KEY (k, t));");
+
+    for (int t = 0; t < N; t++)
+      execute("INSERT INTO %s (k, t, v1, v2) values (?, ?, ?, ?)", "key", t, "v" + t, t + 10);
+
+    for (int i = 0; i < N / 2; i++)
+      execute("DELETE FROM %s WHERE k=? AND t=?", "key", i * 2);
+
+    Object[][] expected = new Object[N/2][];
+    for (int i = 0; i < N / 2; i++)
+    {
+      int t = i * 2 + 1;
+      expected[i] = row("key", t, "v" + t, t + 10);
+    }
+
+    assertRows(execute("SELECT * FROM %s"), expected);
+  }
+
+  @Test
+  public void testRangeTombstones() throws Throwable
+  {
+    int N = 100;
+
+    createTable("CREATE TABLE %s (k text, t1 int, t2 int, v text, PRIMARY KEY (k, t1, t2));");
+
+    for (int t1 = 0; t1 < 3; t1++)
+      for (int t2 = 0; t2 < N; t2++)
+        execute("INSERT INTO %s (k, t1, t2, v) values (?, ?, ?, ?)", "key", t1, t2, "someSemiLargeTextForValue_" + t1 + "_" + t2);
+
+    execute("DELETE FROM %s WHERE k=? AND t1=?", "key", 1);
+
+    Object[][] expected = new Object[2*N][];
+    for (int t2 = 0; t2 < N; t2++)
+    {
+      expected[t2] = row("key", 0, t2, "someSemiLargeTextForValue_0_" + t2);
+      expected[N + t2] = row("key", 2, t2, "someSemiLargeTextForValue_2_" + t2);
+    }
+
+    assertRows(execute("SELECT * FROM %s"), expected);
+  }
+
+  @Test
+  public void test2ndaryIndexes() throws Throwable
+  {
+    createTable("CREATE TABLE %s (k text, t int, v text, PRIMARY KEY (k, t)) WITH transactions = { 'enabled' : true };");
+
+    createIndex("CREATE INDEX %s ON %s(v);");
+
+    execute("INSERT INTO %s (k, t, v) values (?, ?, ?)", "key1", 1, "foo");
+    execute("INSERT INTO %s (k, t, v) values (?, ?, ?)", "key1", 2, "bar");
+    execute("INSERT INTO %s (k, t, v) values (?, ?, ?)", "key2", 1, "foo");
+
+    execute("INSERT INTO %s (k, t, v) values (?, ?, ?)", "key2", 2, "foo");
+    execute("INSERT INTO %s (k, t, v) values (?, ?, ?)", "key2", 3, "bar");
+
+    assertRows(execute("SELECT * FROM %s WHERE v = ?", "foo"),
+      row("key1",  1, "foo"),
+      row("key2",  1, "foo"),
+      row("key2",  2, "foo")
+    );
+
+    assertRows(execute("SELECT * FROM %s WHERE v = ?", "bar"),
+      row("key1",  2, "bar"),
+      row("key2",  3, "bar")
+    );
+  }
+
+  @Test
```

Comment:
Please file a Github issue on this test failure and add `@Ignore ("Disabled for issue #<issue number>")` to temporarily disable this test. We will continue to look into the issue.